tests/unit/src/main/scala/tests/TestingServer.scala

```diff
@@ -1011,17 +1011,56 @@ final class TestingServer(
     }
   }
 
+  private def workspaceDefinitions(
+      getDocumentMarkedWithDefinitions: AbsolutePath => s.TextDocument
+  ): String = {
+    buffers.open.toSeq
+      .sortBy(_.toURI.toString)
+      .map { path =>
+        val textDocument = getDocumentMarkedWithDefinitions(path)
+        val relpath =
+          path.toRelative(workspace).toURI(isDirectory = false).toString
+        val printedTextDocument = Semanticdbs.printTextDocument(textDocument)
+        s"/$relpath\n$printedTextDocument"
+      }
+      .mkString("\n")
+  }
+
   private def toSemanticdbTextDocument(path: AbsolutePath): s.TextDocument = {
+    toDefinitionsTextDocument(path, { params =>
+      server
+        .definitionOrReferences(params, definitionOnly = true)
+        .asJava
+        .get()
+    })
+  }
+
+  private def toTypeDefinitionsTextDocument(
+      path: AbsolutePath
+  ): s.TextDocument = {
+    toDefinitionsTextDocument(path, { params =>
+      server
+        .typeDefinition(params)
+        .asScala
+        .map(DefinitionResult(_, "", None, None))
+        .asJava
+        .get()
+    }, markNoSymbol = false)
+  }
+
+  private def toDefinitionsTextDocument(
+      path: AbsolutePath,
+      getDefinition: TextDocumentPositionParams => DefinitionResult,
+      //in case no resolved symbol gotten
+      markNoSymbol: Boolean = true
```

Comment:
While I wanted to reuse, what `DefinitionLspSuite` does, I was not sure about it testing not actual endpoint method `definition`, which returns a list of locations, but some inner method, which returns a bit richer `DefinitionResult`, to use resolved `symbol` for the tests sake - this a bit breaks the `integration` in testing. So, without the `symbol` I had to tinker this method a bit.